app/experimenter/nimbus-ui/src/hooks/useOutcomes.tsx

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getExperiment_experimentBySlug } from "../types/getExperiment";
+import { useConfig } from "./useConfig";
+
+export function useOutcomes(
+  experiment: getExperiment_experimentBySlug,
+): { primaryOutcomes: OutcomesList; secondaryOutcomes: OutcomesList } {
+  const { outcomes } = useConfig();
+
+  const pairOutcomes = (slugs: OutcomeSlugs) => {
+    if (!slugs || !outcomes) {
+      return [];
+    }
+
+    return slugs
+      .map((slug) => outcomes!.find((outcome) => outcome!.slug === slug))
+      .filter((outcome) => outcome != null);
+  };
+
+  return {
+    primaryOutcomes: pairOutcomes(experiment.primaryOutcomes),
+    secondaryOutcomes: pairOutcomes(experiment.secondaryOutcomes),
+  };
+}
```

Comment:
Since this hook already knows about the experiment and available config outcomes I think in #4752 we should extend it to also return an `available` property with the outcomes applicable to this experiment application.

Reply:
Yeah I was thinking about that but to do that I think you'd have to pass in the experiment you're talking about to the ExperimentConfig call which we're not currently doing?  And that would mostly work for now since we enforce setting application at creation time and then not changing it, but in theory either of those things could change which would affect the semantics of making ExperimentConfig dependent on an experiment id?

Reply:
We might be talking about two separate things, I was talking about returning `available: Outcome[]` from the `useOutcomes` hook. Something like this:

```
const available = outcomes?.filter(
  (outcome) => outcome?.application === experiment.application,
) || [];
```

This is possible because this hook already knows about the experiment (and its application), and the list of outcomes provided by the initial config query.

Make sense?

Reply:
Oh oh you mean entirely on the frontend oh yes totally 👍